File: src/types/content.ts
export interface CmsContent {
  id?: number
  uuid?: string
  slug?: string
  keyword?: string
  title: string
  description?: string
  thumbnail?: string
  editor?: string
  isDeleted?: boolean | null
  createAt?: string
  categoryId?: number | null
}

export type CreateContentPayload = Pick<
  CmsContent,
  'title' | 'slug' | 'keyword' | 'description' | 'thumbnail' | 'editor' | 'categoryId'
>

export type UpdateContentPayload = Pick<
  CmsContent,
  'title' | 'slug' | 'keyword' | 'description' | 'thumbnail' | 'editor' | 'categoryId' | 'isDeleted'
>

File: src/main/java/com/cms/content/model/Content.java
package com.cms.content.model;

import com.cms.category.model.Category;
import com.cms.common.model.BaseEntity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "tb_contents")
@Getter
@Setter
public class Content extends BaseEntity {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

	@Column(unique = true)
	private String uuid;

	@Column(unique = true, columnDefinition = "text")
	private String slug;

	private String keyword;

	@Column(columnDefinition = "text", nullable = false, length = 50)
	private String title;

	@Column(columnDefinition = "text", length = 255)
	private String description;

	@Column(columnDefinition = "text", length = 255)
	private String editor;

	@ManyToOne(fetch = FetchType.LAZY, optional = false)
	@JoinColumn(name = "category_id", nullable = false)
	private Category category;
}
